server/src/services/ranger.service.ts

```diff
@@ -0,0 +1,57 @@
+/* tslint:disable:no-console */
+import { Injectable, RequestTimeoutException } from '@nestjs/common';
+import { DatabaseService } from './db.service';
+import { Ranger } from '../entity/ranger.entity';
+import { User } from "../entity/user.entity";
+
+@Injectable()
+export class RangerService {
+  constructor(private readonly databaseService: DatabaseService) {}
+
+  async addRanger(long: number, lat: number , rangerID: number): Promise<boolean> {
+    const con = await this.databaseService.getConnection();
+    const ranger = new Ranger();
+
+    try {
+        let rangerUser = await JSON.parse( JSON.stringify( await con.getRepository(User).find({id : rangerID, jobType : 'ranger'})));
```

Comment:
No need for JSON.parse(JSON.stringify(...)) - this just deep copies it.

You can do
```
await con.getRepository(User).findOne({id : rangerID, jobType : 'ranger'})
```